src/main/scala/akka/rabbitmq/ConnectionActor.scala

```diff
@@ -59,53 +89,65 @@ class ConnectionActor(
+        case _ =>
+          log.error("{} setup children failed", self.path)
+          self ! Reconnect(connection)
       }
+      stay()
 
-    case Event(msg @ CreateChannel(props, name), Connected(connection)) =>
-      safe(connection.createChannel()) match {
-        case Some(channel) =>
-          val child = newChild(props, name)
-          log.debug("{} creating child {} with channel {}", header(Connected, msg), child, channel)
-          child ! channel
-          stay replying ChannelCreated(child)
-        case None =>
-          val child = newChild(props, name)
-          dropConnectionAndInitiateReconnect(connection)
-          log.debug("{} creating child {} without channel", header(Connected, msg), child)
-          goto(Disconnected) using NoConnection replying ChannelCreated(child)
+    case Event(msg @ Reconnect(oldConnection), Connected(connection)) =>
+      if (oldConnection.getId == connection.getId) {
+        dropConnectionAndNotifyChildren(connection)
+        log.info("{} reconnecting to {} in {}", header(Connected, msg), factory.uri, reconnectionDelay)
+        setTimer(reconnectTimer, Connect, reconnectionDelay, repeat = false)
+        goto(Disconnected) using NoConnection
+      } else {
+        log.debug("{} already reconnected to {}", header(Connected, msg), factory.uri)
+        stay()
       }
 
+    case Event(ProvideChannel, Connected(connection)) =>
+      provideChannel(connection, sender(), ProvideChannel)
+      stay()
+
+    case Event(msg @ CreateChannel(props, name), Connected(connection)) =>
+      val child = newChild(props, name)
+      provideChannel(connection, child, msg)
+      stay replying ChannelCreated(child)
+
     case Event(msg @ AmqpShutdownSignal(cause), Connected(connection)) =>
       // It is important that we check if a shutdown signal pertains to the current connection.
-      // This actor explicitly close the connection ourselves before reconnecting. So if this actor starts to reconnect
-      // in a situation where the connection is still open, it closes it and RabbitMQ will send the actor a
-      // ShutdownSignal. This ShutdownSignal may reach this actor when it already has a new connection, in which case
-      // the actor should of course not act upon it.
       if (msg.appliesTo(connection)) {
         log.debug("{} shutdown (initiated by app {})", header(Connected, msg), cause.isInitiatedByApplication)
-        dropConnectionAndInitiateReconnect(connection)
-        goto(Disconnected) using NoConnection
-      } else stay()
+        self ! Reconnect(connection)
+      }
+      stay()
   }
 
   whenUnhandled {
     case Event(GetState, _) =>
       sender ! stateName
-      stay
+      stay()
+
+    case Event(msg @ DeadLetter(channel: Channel, `self`, child), _) =>
+      log.debug("{} closing channel {} of child {}", header(stateName, msg), channel, child)
+      close(channel)
+      stay()
+
+    case Event(_: DeadLetter, _) =>
```

Comment:
@miriarte I added this case to prevent all kinds of akka warning appearing at run time about unhandled dead letters. Is there maybe a better way you know of to do this (for instance, accept those warnings and stripping away this case?)